libraries/AP_Math/definitions.h

```diff
@@ -117,3 +117,30 @@ static const double WGS84_E = (sqrt(2 * WGS84_F - WGS84_F * WGS84_F));
 
 // Amps microseconds to milliamp hours
 #define AUS_TO_MAH 0.0000002778f
+
+/*
+ * overload fabs so that we get the correct outcome regardless of target type
+ */
+inline float safe_fabs(const float v) {
+    return fabsf(v);
+}
+
+inline double safe_fabs(const double v) {
+#ifdef ALLOW_DOUBLE_MATH_FUNCTIONS
+    return fabs(v);
+#else
+    return fabsf(static_cast<float>(v));
+#endif
+}
+
+inline uint32_t safe_fabs(const uint32_t v) {
```

Comment:
why are we taking an absolute value of an unsigned value? If we are doing that then this could break existing code by effectively removing a cast
for example:
  float x = safe_fabs(7U) / 2;
will give a different answer to:
 float x = fabsf(7U) / 2;

Reply:
ok, I have taken your suggestion of using template specialization and so avoiding the problem